app/routes/cdo/manage/tasks/generic-task-helper.js

```diff
@@ -11,7 +13,7 @@ const taskList = [
   { name: tasks.applicationFeePaid, key: 'record-application-fee-payment', canSubmit: false, label: 'Application fee', apiKey: 'recordApplicationFee', stateKey: 'applicationFeePaid' },
   { name: tasks.form2Sent, key: 'send-form2', canSubmit: false, label: 'Form two', apiKey: 'sendForm2', stateKey: 'form2Sent' },
   { name: tasks.verificationDateRecorded, Model: ViewModelRecordVerificationDates, validation: validateVerificationDates, key: 'submit-form-two', canSubmit: true, label: 'Form 2 confirming dog<br>microchipped and neutered', apiKey: 'submitFormTwo', stateKey: 'verificationDateRecorded' },
-  { name: tasks.microchipDeadlineRecorded, key: 'record-microchip-deadline', canSubmit: false, label: 'When will the dog be fit to be microchipped?', apiKey: 'verifyDates', stateKey: 'verificationDateRecorded' },
+  { name: tasks.microchipDeadlineRecorded, Model: ViewModelRecordMicrochipDeadline, validation: validateMicrochipDeadlineDates, key: 'record-microchip-deadline', canSubmit: true, label: 'When will the dog be fit to be microchipped?', apiKey: 'submitFormTwo', stateKey: 'verificationDateRecorded' },
```

Comment:
Is it correct that both lines 15 and 16 use the same api key of 'submitFormTwo'? Probably yes. I guess line15 api-key isn't used if you end up using the second page of line 16

Reply:
Yes...both need the same key, if there is not a redirect to microchip deadline, then the api is submitted on that action, then the next page also needs to call the api.